apps/remote_control/lib/lexical/remote_control/code_intelligence/definition.ex

```diff
@@ -0,0 +1,76 @@
+defmodule Lexical.RemoteControl.CodeIntelligence.Definition do
+  alias Lexical.SourceFile
+  alias Lexical.SourceFile.Position
+  alias Lexical.SourceFile.Range
+
+  require Logger
+
+  def definition(%SourceFile{} = source_file, %Position{} = position) do
+    maybe_location =
+      source_file
+      |> SourceFile.to_string()
+      |> ElixirSense.definition(position.line + 1, position.character + 1)
+
+    parse_location(maybe_location, source_file)
+  end
+
+  defp parse_location(%ElixirSense.Location{} = location, source_file) do
+    %{file: file, line: line, column: column} = location
+    file_path = file || source_file.path
+
+    case location_to_range(file_path, line, column) do
+      {:ok, {source_file, range}} ->
+        {:ok, %{source_file: source_file, range: range}}
```

Comment:
I prefer to return `{:ok, {source_file, range}}` here, because the result will be `nil` too, now the spec looks like

```elixir
  @spec definition(SourceFile.t(), Position.t()) ::
          {:ok, {SourceFile.t(), Range.t()}} | {:error, String.t()}
  def definition(%SourceFile{} = source_file, %Position{} = position) do
```

Reply:
when will the sourcefile and range be nil?